### PR TITLE
Code generation for convenience layer

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Generate consolidated client.
Generate operation groups.
Moved some shared content to helpers.ts

### CODE REVIEW

File: src/generator/convenience/client.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  text += '\tHTTPClient azcore.Transport\n';
+  text += '\t// LogOptions configures the built-in request logging policy behavior.\n';
+  text += '\tLogOptions azcore.RequestLogOptions\n';
+  text += '\t// Retry configures the built-in retry policy behavior.\n';
+  text += '\tRetry azcore.RetryOptions\n';
+  text += '\t// Telemetry configures the built-in telemetry policy behavior.\n';
+  text += '\tTelemetry azcore.TelemetryOptions\n';
+  text += '}\n\n';
+  text += '// DefaultClientOptions creates a ClientOptions type initialized with default values.\n';
+  text += 'func DefaultClientOptions() ClientOptions {'
```
Comment (catalinaperalta) [inline src/generator/convenience/client.ts]: ;

File: src/generator/convenience/operations.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Session } from '@azure-tools/autorest-extension-base';
+import { camelCase } from '@azure-tools/codegen'
```
Comment (catalinaperalta) [inline src/generator/convenience/operations.ts]: ;

File: src/generator/convenience/client.ts (hunk 1 above)
Comment (jhendrixMSFT) [reply src/generator/convenience/client.ts]: fixed

File: src/generator/convenience/operations.ts (hunk 2 above)
Comment (jhendrixMSFT) [reply src/generator/convenience/operations.ts]: fixed

**Pull request merged**